File: src/types/obs.ts
export type ObsBucket = {
  name: string;
  creation_date?: string | null;
  location?: string | null;
  bucket_type?: string | null;
};

export type ObsListBucketsResponse = {
  buckets?: ObsBucket[];
};

export type ObsObject = {
  key: string;
  last_modified?: string | null;
  etag?: string | null;
  size?: number | null;
  storage_class?: string | null;
};

export type ObsListObjectsResponse = {
  bucket?: string | null;
  prefix?: string | null;
  marker?: string | null;
  next_marker?: string | null;
  is_truncated?: boolean | null;
  objects?: ObsObject[];
};

export type ObsOperationResult = {
  status: string;
  status_code: number;
  body: string;
};

export type ObsGetObjectResult = {
  status: string;
  status_code: number;
  content_base64?: string | null;
  content_type?: string | null;
  body?: string | null;
};

File: src-tauri/src/api/models/obs.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ObsBucket {
    pub name: String,
    pub creation_date: Option<String>,
    pub location: Option<String>,
    pub bucket_type: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ObsListBucketsResponse {
    pub buckets: Vec<ObsBucket>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ObsObject {
    pub key: String,
    pub last_modified: Option<String>,
    pub etag: Option<String>,
    pub size: Option<u64>,
    pub storage_class: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ObsListObjectsResponse {
    pub bucket: String,
    pub prefix: Option<String>,
    pub marker: Option<String>,
    pub next_marker: Option<String>,
    pub is_truncated: bool,
    pub objects: Vec<ObsObject>,
}
